Kitodo/src/main/java/org/kitodo/production/handler/RangeStreamedContentHandler.java

```diff
@@ -0,0 +1,305 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.production.handler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import javax.el.ValueExpression;
+import javax.faces.context.ExternalContext;
+import javax.faces.context.FacesContext;
+import javax.servlet.ServletOutputStream;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.primefaces.application.resource.BaseDynamicContentHandler;
+import org.primefaces.model.StreamedContent;
+import org.primefaces.util.Constants;
+
+/**
+ * This handler allows to stream files e.g. video and audio partially to the browser. Only parts of the file are
+ * requested by the browser are returned. This means, for example, that a video does not have to be loaded completely,
+ * but byte parts are returned as a response of a so-called HTTP range requests.
+ *
+ * <p>For example, in HTML video component you can jump back and forth in the player without the file being completely
+ * loaded or the player is jumping back to the beginning. At the current requested position we preload the file with a
+ * small buffer size.</p>
+ */
+public class RangeStreamedContentHandler extends BaseDynamicContentHandler {
+
+    public static final int DEFAULT_BUFFER_SIZE = 20480; // ..bytes = 20KB.
+
+    private static final String MULTIPART_BOUNDARY = "MULTIPART_BYTERANGES";
+
+    private static final String CLIENT_ABORT_EXCEPTION_CANONICAL_NAME = "org.apache.catalina.connector.ClientAbortException";
+
+    private static final Logger LOGGER = LogManager.getLogger(RangeStreamedContentHandler.class);
```

Comment:
Though it qualifies as a constant, we use lowercase `logger` variable name all through the project